File: app/src/main/java/com/monster/app/niceweather/activities/detail/mvp/DetailUiModel.java
package com.monster.app.niceweather.activities.detail.mvp;

import com.monster.app.niceweather.activities.main.mvp.MainUiModel;
import com.monster.app.niceweather.models.ForecastCityModel;
import com.monster.app.niceweather.models.ForecastModel;

import java.util.List;

/**
 * Created by monster on 3/5/18.
 */

public class DetailUiModel {
    public ForecastCityModel city;
    public List<ForecastModel> data;
    public boolean isLoading;
    public boolean success;
    public String error;

    public DetailUiModel(List<ForecastModel> data, ForecastCityModel city) {
        this.data = data;
        this.success = true;
        this.isLoading = false;
        this.error = "";
        this.city = city;
    }

    public DetailUiModel(boolean isLoading) {
        this.data = null;
        this.success = false;
        this.isLoading = isLoading;
        this.error = "";
        this.city = null;
    }

    public DetailUiModel(String error) {
        this.error = error;
        this.success = false;
        this.isLoading = false;
        this.data = null;
        this.city = null;
    }

    public static DetailUiModel stateLoading() {
        return new DetailUiModel(true);
    }

    public static DetailUiModel stateError(String message) {
        return new DetailUiModel(message);
    }

    public static DetailUiModel stateSuccess(List<ForecastModel> data, ForecastCityModel city) {
        return new DetailUiModel(data, city);
    }
}

File: app/src/main/java/com/monster/app/niceweather/activities/detail/mvp/DetailUiModel.kt
package com.monster.app.niceweather.activities.detail.mvp

import com.monster.app.niceweather.activities.main.mvp.MainUiModel
import com.monster.app.niceweather.models.ForecastCityModel
import com.monster.app.niceweather.models.ForecastModel

/**
 * Created by monster on 3/5/18.
 */

class DetailUiModel {
    var city: ForecastCityModel? = null
    var data: List<ForecastModel>? = null
    var isLoading: Boolean = false
    var success: Boolean = false
    var error: String = ""

    constructor(data: List<ForecastModel>, city: ForecastCityModel) {
        this.data = data
        this.success = true
        this.city = city
    }

    constructor(isLoading: Boolean) {
        this.isLoading = isLoading
    }

    constructor(error: String?) {
        this.error = error ?: ""
    }

    companion object {

        val stateLoading = DetailUiModel(true)

        fun stateError(message: String?) = DetailUiModel(message)

        fun stateSuccess(data: List<ForecastModel>, city: ForecastCityModel) = DetailUiModel(data, city)
    }
}
